src/macros.rs

```diff
@@ -8,15 +8,30 @@
 /// ```
 /// use crab_nbt::nbt;
 ///
-/// let nbt = nbt!("root", {
+/// let key = "key".to_owned();
+/// let value: Vec<u8> = vec![0, 1, 2, 3];
+/// let nbt = nbt!("root nbt_inner name", {
 ///     "float": 1.0,
-///     "key": "value",
-///     "long_array": [L; 1],
-///     "int_array": [Int; 1],
+///     key: "value",
+///     "long_array": [L; 1, 2],
+///     "int_array": [Int; 1, 10, 25],
+///     "byte_array": [B; 0, 1, 0, 0, 1],
 ///     "list": ["a", "b", "c"],
 ///     "nbt_inner": {
 ///         "key": "sub value"
-///     }
+///     },
+///     "byte_array": value,
+/// });
+///
+/// // The macro can also take variables to use data calculated at runtime
+/// let var_as_key = "some_key".to_owned();
+/// let other_key = "other_key".to_owned();
+/// let value = vec![0, 1, 2];
+/// let other_value = vec![3, 2, 1];
+/// let nbt =  nbt!("root", {
+///     var_as_key: "wohoo!",
+///     "the_other_way": value,
+///     other_key: other_value,
 /// });
 /// ```
 ///
```

Comment:
Why do we have this this shouldnt be even in history

Reply:
Why use git when you have comments lmao

Reply:
apparently its documentation throw it out into seperate section like docs folder or something Or maybe use github wiki?

Reply:
okay so my bad didnt know that it appears in rust wiki